### DISCUSSION TITLE
How To Extract Crystallographic Information on Variants

### DISCUSSION THREAD

**Discussion by Savillian** [mtex-toolbox/mtex#1165] [Ask Anything] [answered]
Hi everyone,

Looking through the literature on MTEX, the process to identifying variants in EBSD seems fairly straight forward. However, I've been unable to find any pathway to identify what crystallographic orientation MTEX decides each variant to be.

Is there a way to pull out the crystallographic information (habit plane and direction) of martensite variants (e.g, a list where Variant 1 = (111),[1-10] and so on)? Currently I can't get around the nebulous definition MTEX gives on variant plots or identify a variable with this information stored from the calcVariant process.

Thanks in advance!
Alec

**Comment by nyyssont** [comment]
There is some discussion on the topic here:
https://github.com/mtex-toolbox/mtex/discussions/828

Speaking for martensite specifically, I believe the variant indexing is now by default according to the convention proposed by Morito et al. [1, 2]

After reconstructing, you can get the index number this way:

```
[parentId, fit] = calcParent(childOri,parentRef,p2c,'id')
```

Maybe you are more interested in what the variants mean for titanium phases?

[1] G. Miyamoto, N. Iwata, N. Takayama , T. Furuhara, Acta Materialia, https://doi.org/10.1016/j.actamat.2011.11.018
[2] S. Morito, H. Tanaka, R. Konishi, T. Furuhara, T. Maki, Acta Materialia, https://doi.org/10.1016/S1359-6454(02)00577-3.

**Reply by Savillian** [reply]
Hi nyssont,

Thanks for clarifying. I'm specifically interested in the crystallographic identity of each variant ID for a given reconstruction, regardless of alloy system (steel or titanium) . Knowing the index number (e.g., variant # 4) doesn't tell me the information I'm after, especially if the convention these are listed by as default is for another alloy system entirely (in this case I'm looking at titanium martensite, so the Morito convention wouldn't be applicable). Are there other arguments I could add to the variant calculation process to change this convention, or maybe tweak the source MTEX code?

Ideally I'm looking for a table or variable I can access which lists the crystallographic plane and direction for each martensite variant index number. The ultimate goal here is to relate each variant's orientation to the area transformed for each grain (see if orientation determines favorable variants during transformations). But without the crystallographic information of each variant, I'm up a creek without a paddle.

Thanks again,
Alec

**Comment by frankNiessen** [comment, marked as answer]
Hi Alec, 

MTEX gives you a unique variant numbering to identify the different martensitic orientation variants. You are most likely interested in knowing what these variant numbers mean crystallographically?

In the case of steel, as Tuomo already mentioned, we sorted the variants to conform with the notation in [S. Morito, H. Tanaka, R. Konishi, T. Furuhara, T. Maki, Acta Materialia](https://www-sciencedirect-com.proxy.findit.dtu.dk/science/article/pii/S1359645402005773?via%3Dihub). You can most easily see this by following the [steel example](https://mtex-toolbox.github.io/MaParentGrainReconstruction.html#2) up to the line 
`job.p2c = orientation.KurdjumovSachs(job.csParent, job.csChild)`  and then executing `job.plotVariantPF`. This will plot a pole-figure and label the variants with their variant Ids. You can then compare those Ids with any convention in literature. If you want to change the order of the variants to follow a certain convention, you can swap around the indices in `job.variantMap`. Having those tools you can easily figure out what the variants mean and sort them to your liking. 

For Ti I have often used the convention in [Y.W. Chai, H.Y. Kim, H. Hosoda, S. Miyazaki, Acta Mater. 57 (2009) 4054–4064.](https://www.researchgate.net/publication/248185721_Self-accommodation_in_Ti-Nb_shape_memory_alloys) in Table 1 for my own research, but there is no one single convention for this system as far as I know.

Another hint: You can use `job.parent2ChildInfo` to get some information about the closest rational parallel planes and directions as well as the misorientations between the variants in angle-axis notation. I have just found a bug in this function, so it will only work from the developer-branch right now.

Finally: There is no direct way of determining the habit plane and shear direction from the crystallographic information as you requested in an earlier mail. The only thing you can get is closest parallel plane and direction between the parent and child phases, but this is something completely different. To determine the habit plane and the shear you would need to know the lattice correspondence and some assumptions about the lattice invariant shear. If you want to do such analysis I would recommend you to check out this [PTMC implementation](https://github.com/frankNiessen/PTMC) and this short code on [available work from martensite formation](https://github.com/frankNiessen/martensiteWorkCalculator). You can read about the theory in [F. Niessen, E. V. Pereloma, A.A. Saleh, J. Appl. Crystallogr. 53 (2020).](https://scripts.iucr.org/cgi-bin/paper?S1600576720007451)

I will think about whether there is a simpler way of implementing these things, but often the answer is not as easy as one would like it to be. 

Best wishes
Frank

**Reply by Savillian** [reply]
Hi Frank,

Thank you for the thorough answer and explanation. Little did I know this question would open such a large can of worms, but this does seem appropriate for crystallography. It might take me awhile to digest all this information and the new tools, but I have a much better heading now than I did before.

Thanks for all your help!
Alec

**Reply by nyyssont** [reply]
Alec,

I'd just like to add one thing. Have you taken a look at the section "Parent Child Variants" in the documentation? 

https://mtex-toolbox.github.io/ParentChildVariants.html

It has a rather good explanation of the nuts and bolts of calculating prior orientations.

Specifically:
```
csBeta = crystalSymmetry('432',[3.3 3.3 3.3],'mineral','Ti (beta)');
csAlpha = crystalSymmetry('622',[3 3 4.7],'mineral','Ti (alpha)');
beta2alpha = orientation.map(Miller(1,1,0,csBeta),Miller(0,0,0,1,csAlpha),...
      Miller(-1,1,-1,csBeta),Miller(2,-1,-1,0,csAlpha));
```
This determines the initial misorientation.
```
oriParent = orientation.rand(csBeta);
oriParentSym = oriParent.symmetrise
oriChild = oriParentSym * inv(beta2alpha)
```
When you apply the misorientation to the symmetrically equivalent parent orientations you get a different child each time, although in this particular case some child orientations are also symmetrically equivalent, meaning 12 children total. Using the command "variants" removes these overlapping orientations automatically.

As explained previously, the numbering of the variants is originally based on the order of the symmetrically equivalent parent orientations:
```
oriParentSym = oriParent.symmetrise
```
In the case of martensite, these are changed to follow the Morito convention in a rather simple way. The following is directly from variants.m:
```
  symRot = p2c.CS.properGroup.rot;
  % try to switch to Morito convention
  if length(symRot) == 24
    symRot = symRot([1 17 2 16 3 18 8 22 9 24 7 23 19 11 20 10 21 12 6 15 4 14 5 13]);
  end
```

This would also be the way you could change the ordering to a format meaningful to your specific application. Hope this helps!

-Tuomo

P.S. Frank and others, regarding the habit plane: I understand the cross-section of a martensitic lath should follow the trace of the habit plane on the image plane. If the variant index is determined for each lath following successful reconstruction, do you think it would then be possible to determine the habit plane from a large amount of traces combined with the knowledge of the roughly corresponding austenite and martensite planes and directions? This would be a rather interesting next step following parent phase reconstruction.

**Reply by Savillian** [reply]
Hi Tuomo,

Thanks for the clarification! I've gone through the Parent Child Variants documentation several times piecing together the internal mechanisms and feel pretty good about my variant calculation. I wasn't aware of the ability to manipulate the ordering so easily. I'll look into that more as I continue this deep dive. The only challenge I foresee is not knowing the identity of my variants (there is no convention for my system yet as I'm working with orthorhombic martensite, not Alpha-Ti or Alpha') and what they correspond to in the listing. I can infer this by finding which variants are distributed into packets (common planes) but from there I suspect it might be guesswork to confirm. 

That said, I may be able to bypass this issue by simply calculating the Schmid factor for each variant/packet in each parent grain... That might take some work to develop but I think it's doable (what I thought posting this initial question, so my judgement on such matters may be less than useful).

And regarding your P.S., I think this could work well assuming the system follows a fairly fixed OR. The one I'm working with has a 0.5-2 degree offset which makes reconstructions more challenging (OR Tools provided a "better" OR to complete the reconstruction process), and thus my variant habit planes are slightly off from the expected values. I've been thinking about a way to fix this, but so far I haven't had any successful ideas.

That said, I'll defer judgement on calculating traces to those more in tune with MTEX's inner machinations.

**Comment by frankNiessen** [comment]
Hi Alec,

so you can sort the variants now, that is good, and you say you are missing a convention. There is no clear convention for alpha'' in Ti alloys as there is in steel, but I have seen this convention quite a lot and used it in my work  [Y.W. Chai, H.Y. Kim, H. Hosoda, S. Miyazaki, Acta Mater. 57 (2009) 4054–4064](https://www.researchgate.net/publication/248185721_Self-accommodation_in_Ti-Nb_shape_memory_alloys). In the end of the day it does not matter which convention you use as long as you declare it in your research work (list the 6 alpha'' variants). Of course it is convenient for comparability reasons to stick with previous literature conventions, but it does not matter from a scientific point of view. 


To Tuomo: Yes, in principle it is possible to determine the habit plane this way and I have done this in a recent publication, see Appendix B in [F. Niessen, A.A. Gazder, D.R.G. Mitchell, E. V. Pereloma, Mater. Sci. Eng. A. 802 (2021)](https://www.sciencedirect.com/science/article/pii/S0921509320313022?via%3Dihub), but I am not sure it is very accurate. This is especially true since alpha'' in Ti should be one of the easiest systems for habit plane determination. I am currently working towards an improved version of this together with Azdiar Gazder, but I am not sure it will ever be really good. Anyways, the habit plane cannot be derived from the crystallography itself, it requires the determination by traces or 3D characterization.
